### PR TITLE
Address Xcode 14.3 warnings

### PR DESCRIPTION
**Pull request by mokagio**
I am using Xcode 14.3 locally and seeing a bunch of new warnings. This PR addresses them so folks won't have to.

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

### CODE REVIEW

File: WordPress/Classes/Utility/Logging/SentryStartupEvent.swift [hunk 1]
```diff
@@ -42,7 +42,7 @@ startup time. This will block the thread. Do not use unless you're sure.
                 "Code": error.code,
                 "Description": error.localizedDescription,
                 "User Info": error.userInfo.description
-            ]
+            ] as [String: Any]
```
Comment (mokagio) [inline WordPress/Classes/Utility/Logging/SentryStartupEvent.swift:45]: Screenshot 2023-04-03 at 2 18 52 pm

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Service/BlogDashboardPostsParser.swift [hunk 2]
```diff
@@ -25,7 +25,7 @@ class BlogDashboardPostsParser {
             // If drafts are not synced and the cards API returns zero posts
             // depend on local data
             if blog.dashboardState.draftsSynced || (!posts.hasDrafts && localDraftsCount > 0) {
-                posts["draft"] = Array(repeatElement([:], count: localDraftsCount))
+                posts["draft"] = Array(repeatElement([String: Any](), count: localDraftsCount))
```
Comment (mokagio) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Service/BlogDashboardPostsParser.swift:28]: image

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 3]
```diff
@@ -1293,7 +1293,7 @@ import Combine
         // has to exist within an empty set.
         let predicateForNilTopic = contentType == .saved ?
             NSPredicate(format: "isSavedForLater == YES") :
-            NSPredicate(format: "topic = NULL AND SELF in %@", [])
+            NSPredicate(format: "topic = NULL AND SELF in %@", [String]())
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:1296]: Unrelated, but I don't understand how this predicate works... _I think_ this line is equivalent to `NSPredicate(false)`?

**Review by crazytonyli — APPROVED** [review]

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift (hunk 3 above)
Comment (mokagio) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:1296]: Good question. It surprised me, too. "Self in []"?! 🤔

**Pull request merged**